modules/xfem/src/userobjects/GeometricCut2DUserObject.C

```diff
@@ -0,0 +1,227 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "GeometricCut2DUserObject.h"
+
+// MOOSE includes
+#include "MooseError.h"
+
+// libMesh includes
+#include "libmesh/string_to_enum.h"
+
+// XFEM includes
+#include "XFEMFuncs.h"
+
+template <>
+InputParameters
+validParams<GeometricCut2DUserObject>()
+{
+  // Get input parameters from parent class
+  InputParameters params = validParams<GeometricCutUserObject>();
+
+  // Class description
+  params.addClassDescription("Base class for 2D XFEM Geometric Cut UserObjects");
+  // Return the parameters
+  return params;
+}
+
+GeometricCut2DUserObject::GeometricCut2DUserObject(const InputParameters & parameters)
+  : GeometricCutUserObject(parameters), _cut_line_start_points(), _cut_line_end_points()
+{
+}
+
+GeometricCut2DUserObject::~GeometricCut2DUserObject() {}
+
+bool
+GeometricCut2DUserObject::active(Real time) const
+{
+  bool is_cut_active = false;
+
+  for (unsigned int cut = 0; cut < _cut_line_start_points.size(); ++cut)
+    if (cutFraction(cut, time) > 0)
+      is_cut_active = true;
+
+  return is_cut_active;
+}
+
+bool
+GeometricCut2DUserObject::cutElementByGeometry(const Elem * elem,
+                                               std::vector<CutEdge> & cut_edges,
+                                               Real time) const
+{
+  bool cut_elem = false;
+
+  out << "\n";
```

Comment:
Where is `out` defined? Are you picking this up from the global namespace somehow? `Moose::out`. In any case, if you want to print to the console, use `_console` instead.